### ISSUE TITLE
credential-providers `fromIni` appears broken and/or mis-documented

### ISSUE THREAD

**Issue by djsmedes** [aws/aws-sdk-js-v3#3806]
### Describe the bug

The following code should work:

```typescript
import { fromIni } from "@aws-sdk/credential-providers";

async function logProfile() {
  const foo = fromIni();
  console.log(await foo());
}

logProfile.then(() => console.log("done"))
```

I'm not sure exactly what it would log out, but it's failing with the following error:

```
UnhandledPromiseRejectionWarning: TypeError: (0 , shared_ini_file_loader_1.parseKnownFiles) is not a function
```

Furthermore, the documentation for the `fromIni` function indicates that you can pass in a `profile` argument. However, the actual exported types from the package do not include a `profile` argument in the options type (`FromIniInit`). Doing something like:

```typescript
const foo = fromIni({
    // @ts-ignore
    profile: "some-profile",
  });
```

...might work, but since the function is completely busted, it's not possible to check.

### Expected Behavior

1. `fromIni` function should not crash.
2. `FromIniInit` type should expose `profile` option, and `fromIni` function should respect it.

### Current Behavior

`fromIni` crashed with:

```
UnhandledPromiseRejectionWarning: TypeError: (0 , shared_ini_file_loader_1.parseKnownFiles) is not a function
at /Users/dan/development/ash/node_modules/@aws-sdk/credential-provider-ini/dist-cjs/fromIni.js:7:73
(further stack trace is all my own project files)
```

### Reproduction Steps

1. Use `@aws-sdk/credential-providers` at version 3.131.0 in a project
2. Run a file like the one above

You can even just open the `node` REPL and enter something like:
```javascript
const { fromIni } = require("@aws-sdk/credential-providers")
const foo = fromIni()
foo()
```

### SDK version used

"@aws-sdk/credential-providers": "3.131.0"

### Environment details (OS name and version, etc.)

node v14.19.1 | macOS Monterey 12.4

**Comment by yenfryherrerafeliz** [comment]
Hi @djsmedes, thanks for opening this issue. I tried to reproduce this, but in my environment I got successful results.  Could you please provide any other details that I could be omitting here?.

Please see my code below:
```js
import { fromIni } from "@aws-sdk/credential-providers";

const credentialsProvider = fromIni({
    profile: 'default'
});

console.log(await credentialsProvider());
```
Output:
```console
{
  accessKeyId: 'ACCESS-ID',
  secretAccessKey: 'SECRET-ACCESS-KEY',
  sessionToken: SESSION-TOKEN'
}
```

Also, profile is a valid parameter for fromIni(). We can confirm that by following the inheritance chain of FromIniInit, which is the parameter that fromIni function receives. Please see below:
- [Here](https://github.com/aws/aws-sdk-js-v3/blob/559d71c013f09729128e24ae32a8e0f1090c8520/packages/credential-providers/src/fromIni.ts#L45) fromIni is declared and exported
- [Here](https://github.com/aws/aws-sdk-js-v3/blob/559d71c013f09729128e24ae32a8e0f1090c8520/packages/credential-providers/src/fromIni.ts#L5) we can see that FromIniInit extends from _FromIniInit
- [Here](https://github.com/aws/aws-sdk-js-v3/blob/main/packages/credential-provider-ini/src/fromIni.ts#L8) we can see the definition for _FromIniInit, which extends from SourceProfileInit
- And [here](https://github.com/aws/aws-sdk-js-v3/blob/559d71c013f09729128e24ae32a8e0f1090c8520/packages/shared-ini-file-loader/src/parseKnownFiles.ts#L5) we can see the definition of SourceProfileInit which contains the parameter profile.

Thanks!

**Comment by cmbrehm** [comment]
FWIW I encountered a very similar issue using Cypress.io but it went away after I cleared the Cypress cache and reinstalled.

**Comment by djsmedes** [comment]
Interesting... we also are using Cypress, perhaps that has something to do with it.

I've just tried to trace the import chain on my local filesystem, and it looks like it's somehow resolving to `@aws-sdk/shared-ini-file-loader` at version 3.6.1, which doesn't export `SourceProfileInit` at all. And also doesn't have a file called `parseKnownFIles` which might explain the runtime error.

Odd that it's doing this, our `yarn.lock` shows that the resolved version should be 3.127.0.

I'll try clearing some caches/reinstalling/that sort of thing, and report back.

**Comment by djsmedes** [comment]
There was a nested `node_modules` folder within my install of `@aws-sdk/credential-provider-ini` (at version 3.131.0 according to `package.json`). Within that node modules folder, was a subfolder for package `@aws-sdk/shared-ini-file-loader`, and THAT package's `package.json` showed version 3.6.1, and inspecting the source code, it was clearly way off from current state of that repo -- which makes sense as that would put it over 120 versions behind. I simply deleted that folder (so from my repo root, `./node_modules/@aws-sdk/credential-provider-ini/node_modules/@aws-sdk/credential-provider-ini/shared-ini-file-loader`), and now everything works fine.

No idea how or why that got there, but I'm guessing it's the fault of `yarn` or maybe even Cypress, so I'll close this issue.

Thank you very much for the speedy reply!
